indra/newview/lltooldraganddrop.cpp

```diff
@@ -1077,51 +1077,71 @@ void set_texture_to_material(LLViewerObject* hit_obj,
                              LLGLTFMaterial::TextureInfo drop_channel)
 {
     LLTextureEntry* te = hit_obj->getTE(hit_face);
-    if (te)
+    if (!te)
     {
-        LLPointer<LLGLTFMaterial> material = te->getGLTFMaterialOverride();
+        return;
+    }
 
-        // make a copy to not invalidate existing
-        // material for multiple objects
-        if (material.isNull())
-        {
-            // Start with a material override which does not make any changes
-            material = new LLGLTFMaterial();
-        }
-        else
-        {
-            material = new LLGLTFMaterial(*material);
-        }
+    const LLUUID base_mat_id = hit_obj->getRenderMaterialID(hit_face);
+    if (base_mat_id.isNull())
+    {
+        return;
+    }
 
-        switch (drop_channel)
-        {
-            case LLGLTFMaterial::GLTF_TEXTURE_INFO_BASE_COLOR:
-            default:
-                {
-                    material->setBaseColorId(asset_id);
-                }
-                break;
+    if (hit_obj->isInventoryDirty() && hit_obj->hasInventoryListeners())
+    {
+        hit_obj->requestInventory();
+        return;
```

Comment:
Should this just fall through or cancel? We do the same check when applying limited permission items, so it isn't without precendent.